tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9760,198 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+		})
+		stepStr = "Test beyond max drives per node"
+		Step(stepStr, func() {
+			log.InfoD(stepStr)
+			var selectedPools string
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			//first add all the new pools
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			//get the number of available drives in the node
+			drvNum, err := Inst().N.RunCommand(selectedNode, "lsblk -l -d -e 11 -n -o NAME|wc -l", node.ConnectionOpts{
+				Timeout:         defaultCommandTimeout,
+				TimeBeforeRetry: defaultCommandRetry,
+				IgnoreError:     false,
+				Sudo:            false,
+			})
+			drvNum1 := strings.TrimSpace(string(drvNum))
+			drvNum1 = strings.Trim(drvNum1, "\n")
+			drvNumInt, err := strconv.Atoi(drvNum1)
+			log.FailOnError(err, "failed to convert to int")
+			maxDriveLimit, err := GetPoolMaxCloudDriveLimit(&selectedNode)
+			log.FailOnError(err, "failed to get pool drive limit")
+			maxDriveLimitInt := int(maxDriveLimit)
+			maxDriveAllowed := maxDriveLimitInt - drvNumInt
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			//expand by adding disk
+			selectedNodeForOps, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting node")
+			poolListForOps, err := GetPoolsDetailsOnNode(selectedNodeForOps)
+			log.FailOnError(err, "failed while getting updated node pool list")
+
+			//add max drive allowed per node
+			for maxDriveAllowed > 0 {
+				for i := 0; i < len(poolListForOps); i++ {
+					err = Inst().V.RefreshDriverEndpoints()
+					log.FailOnError(err, "error refreshing volume endpoints")
+					drvSize, err := getPoolDiskSize(poolListForOps[i])
+					log.FailOnError(err, "error getting drive size for pool [%s]", poolListForOps[i].Uuid)
+					drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+					log.FailOnError(err, "error in getting pool with IO with error")
+					if drvs, ok := drvMap[fmt.Sprintf("%d", poolListForOps[i].ID)]; ok {
+						if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+							drivesThatCanBeAdded := (POOL_MAX_CLOUD_DRIVES - len(drvs))
+							for j := 1; j <= drivesThatCanBeAdded; j++ {
+								driveSize := drvSize * uint64(j)
+								expectedSize := (poolListForOps[i].TotalSize / units.GiB) + driveSize
+								isjournal, err := isJournalEnabled()
+								log.FailOnError(err, "Failed to check is journal enabled")
+								expectedSizeWithJournal := expectedSize
+								if isjournal {
+									journalSize := uint64(3)
+									expectedSizeWithJournal = expectedSizeWithJournal - journalSize
+								}
+								err = Inst().V.ExpandPool(poolListForOps[i].Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+								log.FailOnError(err, "error while expanding pool")
+								maxDriveAllowed = maxDriveAllowed - 1
+								if maxDriveAllowed == 0 {
+									break
+								}
```

Comment:
if pool expansion is triggered successfully and maxDriveAllowed is 0, waitForPoolToBeResized will not be called.
plz check

Reply:
done made the changes